File: 3136_validWord.cpp
// https://leetcode.com/problems/valid-word/

#include<iostream>
#include<vector>
#include<algorithm>
#include<list>
#include<stack>
#include<stdlib.h>
using namespace std;

class Solution {
public:
    bool isValid(string word) 
    {
        if(word.length() < 3) return false;
        
        int vow = 0, cons = 0;
        for(int i = 0; i < word.length(); i++)
        {
            if(!isalnum(word[i])) return false;

            if(tolower(word[i]) == 'a' || tolower(word[i]) == 'e' || tolower(word[i]) == 'i' || tolower(word[i]) == 'o' || tolower(word[i]) == 'u') vow++;

            else if(isalpha(word[i])) cons++;
        }
        return ((vow > 0) && (cons > 0));
    }
};

int main(){
    system("cls");
    string word = "234$das";
    Solution S;
    cout << S.isValid(word) << endl;
    return 0;
}

File: 3136_validWord.py
# https://leetcode.com/problems/valid-word/
# A word is considered valid if:
    # It contains a minimum of 3 characters.
    # It contains only digits (0-9), and English letters (uppercase and lowercase).
    # It includes at least one vowel.
    # It includes at least one consonant.

# You are given a string word.
# Return true if word is valid, otherwise, return false.

class Solution:
    def isValid(self, word: str) -> bool:
        if len(word) < 3:
            return False
        
        vow, cons = 0, 0
        for ch in word:
            if ch.isalnum() == False:
                return False

            if ch in 'aeiouAEIOU':
                vow += 1

            elif ch.isalpha() and ch not in 'aeiouAEIOU':
                cons += 1

        return (vow > 0) and (cons > 0)

s = "234adas"
S = Solution()
print(S.isValid(s))